crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs

```diff
@@ -405,14 +405,27 @@ impl Analyzer<'_, '_> {
                             mark_var_as_truthy = true;
                             ty
                         })
-                        .map_err(|v| {
-                            match v.actual() {
-                                Error::CannotAssignToNonVariable { .. } => {
-                                    skip_right = true;
+                        .convert_err(|err| {
+                            skip_right = true;
+                            match err.actual() {
+                                Error::CannotAssignToNonVariable { .. } => match analyzer.scope.vars.get(&i.into()) {
+                                    Some(v) if v.kind == VarKind::Fn => Error::CannotAssignToFunction { span },
+                                    _ => err,
+                                },
+                                Error::NotVariable { ty, .. } => match ty {
+                                    Some(ty) => match **ty {
```

Comment:
Please change this to `match ty.normalize() { ... }`

Reply:
In this way we can avoid cloning

Reply:
✅ [`43a0b63` (#192)](https://github.com/dudykr/stc/pull/192/commits/43a0b63877b6837b2505f8695264226b0c134fdc)